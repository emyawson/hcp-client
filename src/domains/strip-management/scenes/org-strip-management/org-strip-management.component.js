import React from 'react';

import {
  Tabs,
  Card,
  GridContainer,
  GridItem,
  TabsContent,
  TabsHeader,
  TabLinkItem,
} from 'src/components';
import { translate } from 'src/i18n';

export const OrgStripManagement = ({ match, children }) => (
  <GridContainer marginBottom>
    <GridItem span="12">
      <Card>
        <Tabs>
          <TabsHeader mb={4} mr="auto" ml="0">
            <TabLinkItem name={translate('orgStock.title')} path="" />
          </TabsHeader>
          <TabsContent>{children}</TabsContent>
        </Tabs>
      </Card>
    </GridItem>
  </GridContainer>
);
