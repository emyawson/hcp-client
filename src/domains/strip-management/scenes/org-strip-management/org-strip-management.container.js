import { compose } from 'recompose';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { mapDispatchers } from 'src/utils';
import { withNavigators } from 'src/utils/with-navigators';
import { getPatientStockRequest } from 'src/core';

import { OrgStripManagement } from './org-strip-management.component';
import { orgStripManagementConnector } from './store/org-strip-management.selector';

const dispatchers = mapDispatchers({
  getPatientStockRequest: getPatientStockRequest.start,
});

export const addOrgStripManagement = compose(
  connect(
    orgStripManagementConnector,
    dispatchers,
  ),
);

export const OrgStripManagementWrapper = compose(
  withNavigators({
    hasLeftNav: true,
    hasTopNav: true,
  }),
  withRouter,
  addOrgStripManagement,
)(OrgStripManagement);
