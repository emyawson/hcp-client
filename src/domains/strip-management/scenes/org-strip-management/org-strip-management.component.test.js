import React from 'react';
import { shallow } from 'enzyme';

import { OrgStripManagement } from './org-strip-management.component';

describe('Organization Strip Management component', () => {
  it('renders correctly', () => {
    const props = {
      match: {
        params: {
          id: 22,
        },
      },
    };
    const component = shallow(<OrgStripManagement {...props} />);
    expect(component).toMatchSnapshot();
  });
});
