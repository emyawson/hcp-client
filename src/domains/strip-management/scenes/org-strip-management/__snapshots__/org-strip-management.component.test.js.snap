// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization Strip Management component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    match={
      Object {
        "params": Object {
          "id": 22,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <GridItem
        span="12"
      >
        <Card>
          <Unknown>
            <TabsHeader
              mb={4}
              ml="0"
              mr="auto"
            >
              <withRouter()
                name="Stock Details"
                path=""
              />
            </TabsHeader>
            <Unknown />
          </Unknown>
        </Card>
      </GridItem>,
      "marginBottom": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Card>
          <Unknown>
            <TabsHeader
              mb={4}
              ml="0"
              mr="auto"
            >
              <withRouter()
                name="Stock Details"
                path=""
              />
            </TabsHeader>
            <Unknown />
          </Unknown>
        </Card>,
        "span": "12",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <Unknown>
            <TabsHeader
              mb={4}
              ml="0"
              mr="auto"
            >
              <withRouter()
                name="Stock Details"
                path=""
              />
            </TabsHeader>
            <Unknown />
          </Unknown>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <TabsHeader
                mb={4}
                ml="0"
                mr="auto"
              >
                <withRouter()
                  name="Stock Details"
                  path=""
                />
              </TabsHeader>,
              <Unknown />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": <withRouter()
                  name="Stock Details"
                  path=""
                />,
                "mb": 4,
                "ml": "0",
                "mr": "auto",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "name": "Stock Details",
                  "path": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <GridItem
          span="12"
        >
          <Card>
            <Unknown>
              <TabsHeader
                mb={4}
                ml="0"
                mr="auto"
              >
                <withRouter()
                  name="Stock Details"
                  path=""
                />
              </TabsHeader>
              <Unknown />
            </Unknown>
          </Card>
        </GridItem>,
        "marginBottom": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Card>
            <Unknown>
              <TabsHeader
                mb={4}
                ml="0"
                mr="auto"
              >
                <withRouter()
                  name="Stock Details"
                  path=""
                />
              </TabsHeader>
              <Unknown />
            </Unknown>
          </Card>,
          "span": "12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <Unknown>
              <TabsHeader
                mb={4}
                ml="0"
                mr="auto"
              >
                <withRouter()
                  name="Stock Details"
                  path=""
                />
              </TabsHeader>
              <Unknown />
            </Unknown>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <TabsHeader
                  mb={4}
                  ml="0"
                  mr="auto"
                >
                  <withRouter()
                    name="Stock Details"
                    path=""
                  />
                </TabsHeader>,
                <Unknown />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": <withRouter()
                    name="Stock Details"
                    path=""
                  />,
                  "mb": 4,
                  "ml": "0",
                  "mr": "auto",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "name": "Stock Details",
                    "path": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
